### PR TITLE
Improve filename parsing in generic parsers

### PR DESCRIPTION
**Pull request by illusional**
- Better handling of splitting filenames (on commas and semicolons)
- Refactor splitting and cleaning up filenames for reuse and consistency
- Warning of spaces in filenames (possible source of error in case they expected space separated filenames)
- Ensure the file map validation parses read filenames by calling read_filenames (as opposed to reimplementing this check)
- Bump version patch commit

### CODE REVIEW

File: sample_metadata/parser/generic_metadata_parser.py [hunk 1]
```diff
@@ -268,21 +275,22 @@ async def validate_sample_map(self, sample_map: Dict[str, List[Dict[str, Any]]])
...
     ) -> List[str]:
         """
         Check that the files in the search_paths are completely covered by the sample_map
         """
-        filenames = []
+        filename_promises = []
         for grp in rows:
             for r in grp if isinstance(grp, list) else [grp]:
-                filenames.extend(r.get(self.reads_column, '').split(','))
+                filename_promises.append(self.get_read_filenames('TEST', r))
```
Comment (violetbrina) [inline sample_metadata/parser/generic_metadata_parser.py:291]: Should this 'TEST' be here?

**Review by violetbrina — CHANGES_REQUESTED** [review]
Looks good, just one comment to address.

File: sample_metadata/parser/generic_metadata_parser.py (hunk 1 above)
Comment (illusional) [reply sample_metadata/parser/generic_metadata_parser.py]: Thank you! I initially thought it wouldn't matter, but there's an easy fix anyway, addressed :)

**Review by violetbrina — APPROVED** [review]
Looks good!

**Pull request merged**